core-kotlin-modules/core-kotlin-collections-6/src/test/kotlin/com/baeldung/parallelOperationsCollections/ParallelOperationCollectionsUnitTest.kt

```diff
@@ -0,0 +1,209 @@
+package com.baeldung.parallelOperationsCollections
+
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.toObservable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.slf4j.LoggerFactory
+import java.time.Duration
+import java.time.Instant
+import java.util.concurrent.Callable
+import java.util.concurrent.Executors
+import java.util.stream.Collectors
+
+
+class ParallelOperationCollectionsUnitTest {
+
+    private val logger = LoggerFactory.getLogger("")
+
+    data class Person(val name: String, val age: Int, var isAdult: Boolean? = null)
+
+    private val people = listOf(
+        Person("Martin", 12),
+        Person("Ahmad", 42),
+        Person("Alina", 13),
+        Person("Alice", 30),
+        Person("Bob", 16),
+        Person("Charlie", 40)
+    )
+
+    private fun List<Person>.assertResultsTrue() {
+        assertThat(this).containsExactly(
+            Person("Bob", 16, false),
+            Person("Alice", 30, true),
+            Person("Charlie", 40, true),
+            Person("Ahmad", 42, true)
+        )
+    }
+
+    private fun String.printAsHeader() {
+        logger.info("{} {}", "-".repeat(32 - Thread.currentThread().name.length), this)
+    }
+
+    private fun Person.setAdult() {
+        this.isAdult = this.age >= 18
+        logger.info("{} {}", " ".repeat(32 - Thread.currentThread().name.length), this)
+    }
+
+    private fun Instant.printTotalTime() {
+        val totalTime = Duration.between(this, Instant.now()).toMillis()
+        logger.info(
+            "{} Total time taken: {} ms \n", "-".repeat(32 - Thread.currentThread().name.length), totalTime
+        )
+    }
```

Comment:
I don't think we really need the .repeat, can we simplify as:
```suggestion
    private fun String.printAsHeader() {
        logger.info(this)
    }

    private fun Person.setAdult() {
        this.isAdult = this.age >= 18
        logger.info(this.toString())
    }
    
    private fun Instant.printTotalTime() {
        val totalTime = Duration.between(this, Instant.now()).toMillis()
        logger.info("Total time taken: {} ms", totalTime)
    }
```